src/main/java/io/confluent/connect/jdbc/sink/JdbcSinkConfig.java

```diff
@@ -102,6 +102,13 @@
       + " table, when possible.";
   private static final String BATCH_SIZE_DISPLAY = "Batch Size";
 
+  public static final String DELETE_ENABLED = "delete.enabled";
+  private static final String DELETE_ENABLED_DEFAULT = "false";
+  private static final String DELETE_ENABLED_DOC =
+      "Whether to treat ``null`` record values as deletes. Requires ``pk.mode`` "
+      + "to be ``record_key``.";
```

Comment:
Because `Validator` only has the context of an individual configuration value, it's not straightforward to validate  one config value (e.g., `pk.mode` must be `record_key`). One technique that works, however, is to add a check in the `JdbcSinkConfig` constructor to check these two config values against this constraint and throw `ConfigException` if they are not compatible. This will help ensure that invalid configurations are caught immediately before the connector is started and without having to figure out why the task failing at a later time.

Also, to help C3 and other GUI tools present valid options to begin with, how about adding
* a `Recommender` implementation for `delete.enabled` that hides it if `pk.mode` is not `record_key`; and
* a `Recommender` for `pk.mode` that recommends only `record_key` if `delete.enabled=true`.

It also may be

Reply:
added recommenders for both and tested locally in C3